Framework/DataHandling/src/SaveGSS.cpp

```diff
@@ -708,38 +715,41 @@ void SaveGSS::setOtherProperties(IAlgorithm *alg,
   *
   * @throws :: If for any reason we cannot run the algorithm
   */
-void SaveGSS::validateUserInput() const {
+std::map<std::string, std::string> SaveGSS::validateInputs() {
+  std::map<std::string, std::string> result;
+
+  API::MatrixWorkspace_const_sptr input_ws = getProperty("InputWorkspace");
+
   // Check the number of histogram/spectra < 99
-  const auto nHist = static_cast<int>(m_inputWS->getNumberHistograms());
+  const auto nHist = static_cast<int>(input_ws->getNumberHistograms());
   const bool split = getProperty("SplitFiles");
   if (nHist > 99 && !split) {
     std::string outError = "Number of Spectra(" + std::to_string(nHist) +
                            ") cannot be larger than 99 for GSAS file";
-    g_log.error(outError);
-    throw std::invalid_argument(outError);
+    result["InputWorkspace"] = outError;
+    result["SplitFiles"] = outError;
+    return result;
   }
 
   // Check we have any output filenames
-  assert(m_outFileNames.size() > 0);
-
-  const bool append = getProperty("Append");
-  for (const auto &filename : m_outFileNames) {
-    if (!append && doesFileExist(filename)) {
-      g_log.warning("Target GSAS file " + filename +
-                    " exists and will be overwritten.\n");
-    } else if (append && !doesFileExist(filename)) {
-      g_log.warning("Target GSAS file " + filename +
-                    " does not exist but algorithm was set to append.\n");
-    }
+  std::string output_file_name = getProperty("Filename");
```

Comment:
I think this is already done for you by the property when you say it is `API::FileProperty::Save`.